isso/db/comments.py

```diff
@@ -217,16 +217,16 @@ def fetchall(self, mode=5, after=0, parent='any', order_by='id',
 
         # custom sanitization
         if order_by not in ['id', 'created', 'modified', 'likes', 'dislikes', 'tid']:
-            sql.append('ORDER BY ')
-            sql.append("comments.created")
+            sql.append('ORDER BY CASE WHEN comments.parent IS NOT NULL THEN comments.id END, ')
+            sql.append('comments.created')
             if not asc:
                 sql.append(' DESC')
         else:
-            sql.append('ORDER BY ')
+            sql.append('ORDER BY CASE WHEN comments.parent IS NOT NULL THEN comments.id END, ')
```

Comment:
Please use `comments.created` instead of `comments.id` for sorting. The `id` field is the (auto-incrementing) primary key. A PK id might be reassigned when a comment is deleted and a subsequently created comment grabs the next available id, shuffling the sort order.